libs/full/async_cuda/CMakeLists.txt

```diff
@@ -57,6 +63,6 @@ add_hpx_module(
   COMPAT_HEADERS ${async_cuda_compat_headers}
   MODULE_DEPENDENCIES hpx_naming hpx_program_options hpx_runtime_configuration
                       hpx_runtime_local ${async_cuda_extra_module_dependencies}
-  DEPENDENCIES hpx_core hpx_parallelism Cuda::cuda
+  DEPENDENCIES hpx_core hpx_parallelism Cuda::cuda ${async_cuda_extra_deps}
```

Comment:
Should `Cuda::cuda` be there all the time? I'd have expected that target not to be defined with HIP enabled. Do you know why CMake doesn't complain about it?

Reply:
It shouldn't be there I thought I separated it already, it doesn't complain cause it looks like `HPX_WITH_CUDA` is not checked in `HPX_SetupCUDA.cmake` so when I

Reply:
For now, `HPX_WITH_CUDA` implies `HPX_WITH_COMPUTE_CUDA` and `HPX_WITH_ASYNC_CUDA`, it shouldn't be true anymore, so I will remove this implication in the `CMakeLists.txt` should I add a `HPX_WITH_GPU` which would enable both sub-options knowing those will be renamed in another PR with `HPX_WITH_COMPUTE_GPU` and `HPX_WITH_ASYNC_GPU` or similar?